src/view/films-list.js

```diff
@@ -0,0 +1,9 @@
+export const createFilmsListTemplate = ( title, isExtra ) => (`
+  <section class="films-list ${isExtra ? 'films-list--extra' : ''}">
```

Comment:
Раздражает, что тут создается в коде пробел после класса в том случае, когда модификатор не нужен.

Можно, конечно, написать так:
```suggestion
  <section class="films-list${isExtra ? ' films-list--extra' : ''}">
```

Но это тупость какая-то. HTML будет аккуратнее, зато JS стремный.
Есть какое-то красивое решение? Или это вообще не важно и на форматирование можно забить?

Reply:
> Есть какое-то красивое решение?

<details>
<summary>Пример некрасивого</summary>

```js
// utils
const mapStyles = (style) => {
  if (typeof style === 'string') {
    return style;
  }
  if (Array.isArray(style)) {
    return style.map(mapStyles).join('').trim();
  }
};

const formatStyle = (...styles) => styles.map(mapStyles).join(' ').trimEnd();
```
```js
// view/films-list.js
const createFilmsListTemplate = (title, { isExtra = false } = {}) => (
  `<section class="${formatStyle`films-list ${isExtra && 'films-list--extra'}`}">
    <h2 class="${formatStyle`films-list__title ${isExtra && 'visually-hidden'}`}">${title}</h2>
  </section>`
);

createFilmsListTemplate('All movies', { isExtra: false });
createFilmsListTemplate('Top rated', { isExtra: true });
```
```html
<section class="films-list">
    <h2 class="films-list__title">All movies</h2>
  </section>

<section class="films-list films-list--extra">
    <h2 class="films-list__title visually-hidden">Top rated</h2>
  </section>
```
</details>

> Или это вообще не важно и на форматирование можно забить?

Для нас это не важно и можно забить 😃

Reply:
_Мы изучим ваш код и вернемся с ответом через неделю_

Как же мне теперь забить, если вон код с решением, и я не смогу спать, пока не пойму что там происходит 😀